### PR TITLE
Add cmdline switch to convert deprecation warnings to errors

### PR DESCRIPTION
**Pull request by jakebolewski**
This makes it easier to catch mistaken use of deprecated functionality in test suites.

### CODE REVIEW

File: doc/manual/getting-started.rst [hunk 1]
```diff
@@ -140,7 +140,8 @@ those available for the ``perl`` and ``ruby`` programs::
                               Run time-intensive code optimizations
     --check-bounds={yes|no}   Emit bounds checks always or never (ignoring declarations)
     --dump-bitcode={yes|no}   Dump bitcode for the system image (used with --build)
-    --depwarn={yes|no}        Enable or disable syntax and method deprecation warnings
+    --depwarn={yes|no}        Enable or disable syntax and method deprecation warnings ('error' turns warnings into errors)
```
Comment (joehuchette) [inline doc/manual/getting-started.rst]: ``--depwarn={yes|no|error}``?

File: ui/repl.c [hunk 2]
```diff
@@ -84,7 +84,7 @@ static const char opts[]  =
     " --math-mode={ieee,fast}   Disallow or enable unsafe floating point optimizations (overrides @fastmath declaration)\n\n"
 
     // error and warning options
-    " --depwarn={yes|no}        Enable or disable syntax and method deprecation warnings\n\n"
+    " --depwarn={yes|no}        Enable or disable syntax and method deprecation warnings ('error' turns warnings into errors)\n\n"
```
Comment (yuyichao) [inline ui/repl.c]: Also here?

File: doc/manual/getting-started.rst (hunk 1 above)
Comment (jakebolewski) [inline doc/manual/getting-started.rst]: Thanks.

**Comment by tkelman** [comment]
I like this. Okay to merge?

**Pull request merged**